db/version_builder.h

```diff
@@ -44,5 +46,21 @@ class VersionBuilder {
   Rep* rep_;
 };
 
+// A wrapper of version builder which references the current version in
+// constructor and unref it in the destructor.
+// Both of the constructor and destructor need to be called inside DB Mutex.
+class BaseReferencedVersionBuilder {
+ public:
+  explicit BaseReferencedVersionBuilder(ColumnFamilyData* cfd);
+  BaseReferencedVersionBuilder(ColumnFamilyData* cfd, Version* v);
+  ~BaseReferencedVersionBuilder();
+  VersionBuilder* version_builder() const { return version_builder_.get(); }
+
+ private:
+  std::unique_ptr<VersionBuilder> version_builder_;
+  Version* version_;
+  const bool should_ref_cfd_version_;
```

Comment:
I suggest adding a comment to clarify what this is used for.

Reply:
I am actually rethinking this. Maybe I can try to remove this variable. If it's non-trivial, will add a comment.

Reply:
Removed, thus no need for comment any more.